### PR TITLE
Test

### PR DESCRIPTION
**Pull request by evanstade**
Add bucket durability and IndexedDB text.

***
<a href="https://pr-preview.s3.amazonaws.com/evanstade/storage-buckets/pull/45.html" title="Last updated on Dec 22, 2022, 2:10 AM UTC (df202ba)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/storage-buckets/45/df01949...evanstade:df202ba.html" title="Last updated on Dec 22, 2022, 2:10 AM UTC (df202ba)">Diff</a>

### CODE REVIEW

**Comment by evanstade** [comment]
ptal. Note the variable durability verbiage:

- IDB durability MAY be respected (see idb spec)
- StorageBucketOptions durability MAY be respected when creating the bucket
- Bucket's durability MUST be respected (i.e. should reflect what the user agent is going to do)

File: index.bs [hunk 1]
```diff
@@ -52,6 +52,12 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+When not null, {{StorageBucketOptions/durability}} is a hint to the user agent
+specifying the desired {{StorageBucket/durability}}. The user agent MAY
```
Comment (ayuishii) [inline index.bs:57]: Is it standard to make these all caps? I've never noticed this pattern before but my spec knowledge is lacking.

File: index.bs [hunk 2]
```diff
@@ -78,4 +84,32 @@ interface StorageBucket {
...
 </xmp>
 
+<h3 id="storage-bucket-durability">Durability</h3>
+
+A {{StorageBucket}} has a <dfn>bucket durability</dfn>. This reflects whether the user agent
+will prioritize performance or durability when completing operations on data in the bucket.
+The value is one of the following:
+
+: "{{StorageBucketDurability/strict}}"
+:: The user agent MUST consider that operations on data in the bucket are successful only after verifying that outstanding changes have been written to a persistent storage medium.
```
Comment (ayuishii) [inline index.bs:94]: If IDB transactions are `may` should this match this as well? If the IDB transaction durability is only a hint, wouldn't this only be able to promise the same guarantee?

File: index.bs [hunk 3]
```diff
@@ -52,6 +52,12 @@ dictionary StorageBucketOptions {
 </xmp>
 
 <h3 id="storage-bucket-open">Creating a bucket</h3>
+
+When not null, {{StorageBucketOptions/durability}} is a hint to the user agent
+specifying the desired {{StorageBucket/durability}}. The user agent MAY
+create a new {{StorageBucket}} with this [=bucket durability=]. The user agent
```
Comment (ayuishii) [inline index.bs:58]: Should this also include the default transaction behavior? [1] Idk if we want to further discuss how it should behave separately given [2].

[1] https://github.com/WICG/storage-buckets/blob/gh-pages/explainer.md#default-to-strict-durability
[2] https://github.com/WICG/storage-buckets/issues/3

**Review by ayuishii — COMMENTED** [review]
Thanks for the change! Overall LGTM, just a few questions.

File: index.bs (hunk 1 above)
Comment (mkruisselbrink) [reply index.bs:57]: Historically more w3c specs used all-caps for these. https://infra.spec.whatwg.org/#conformance now says that using lower-case is fine as well, despite that being a violation of some RFC (although not sure it is linking to the right RFC).

File: index.bs (hunk 2 above)
Comment (evanstade) [reply index.bs:94]: The options that are passed during bucket creation may or may not be respected, however the StorageBucket property must report what the UA is actually going to do.

File: index.bs (hunk 3 above)
Comment (evanstade) [reply index.bs:58]: To follow in the footsteps of IDB, I don't think we should specify a default. A UA can be spec-conformant without any support for "strict" or any support for "relaxed", so it's not possible to enforce a default.

File: index.bs (hunk 1 above)
Comment (ayuishii) [reply index.bs:57]: Interesting, thanks for the historical context! Will leave it up to Evan then.

**Review by ayuishii — APPROVED** [review]
LGTM

File: index.bs [hunk 4]
```diff
@@ -78,4 +84,32 @@ interface StorageBucket {
...
+:: The user agent MUST consider that operations on data in the bucket are successful as soon as all outstanding changes have been written to the operating system, without subsequent verification.
+
+The user agent MUST ignore the [=bucket durability=] for operations that otherwise override default durability behavior.
+
+<h3 id="storage-bucket-indexeddb">Creating an Indexed Database</h3>
+
+Each {{StorageBucket}} has an {{IDBFactory}} object. {{IDBDatabase}} objects created by
+{{StorageBucket/indexedDB}} are associated with the bucket.
+
+The user agent should consider the associated {{StorageBucket/durability}} when evaluating the <a spec="IndexedDB">durability hint</a> |durability|
```
Comment (ayuishii) [inline index.bs:105]: Oh, just noticed. Should this be surrounded by `<div algorithm></div>`?

Seems like a common pattern I see when explaining method steps ex [1].

[1] https://github.com/w3c/IndexedDB/blob/main/index.bs#L1114

**Pull request closed without merging**